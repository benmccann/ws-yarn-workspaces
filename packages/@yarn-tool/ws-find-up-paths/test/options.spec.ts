//@noUnusedParameters:false

import { handleOptions } from '../index';

beforeAll(async () =>
{

});

const baseExpectedRuntimeBase = {
	cwd: expect.any(String),
	stopPath: expect.any(Array),
}

const baseExpectedRuntime = {
	...baseExpectedRuntimeBase,
	opts: {
		...baseExpectedRuntimeBase,
	},
	path: expect.anything(),
	rootData: {
		cwd: expect.any(String),
		pkg: expect.any(String),
		ws: expect.any(String),
		root: expect.any(String),
	},
}

test(`ignoreCurrentPackage`, () =>
{

	let actual = handleOptions({
		ignoreCurrentPackage: true,
	});

	expect(actual).toMatchSnapshot(baseExpectedRuntime);

	expect(actual).toHaveProperty('opts.ignoreCurrentPackage', true);
	expect(actual).toHaveProperty('opts.includeCurrentDirectory', false);

});
