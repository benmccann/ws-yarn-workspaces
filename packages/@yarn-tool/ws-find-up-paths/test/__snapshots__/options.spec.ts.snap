// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ignoreCurrentPackage 1`] = `
{
  "cwd": Any<String>,
  "limit": Infinity,
  "opts": {
    "cwd": Any<String>,
    "ignoreCurrentPackage": true,
    "includeCurrentDirectory": false,
    "limit": Infinity,
    "stopPath": Any<Array>,
  },
  "path": Anything,
  "rootData": {
    "cwd": Any<String>,
    "hasWorkspace": true,
    "isRoot": false,
    "isWorkspace": false,
    "pkg": Any<String>,
    "root": Any<String>,
    "ws": Any<String>,
  },
  "stopPath": Any<Array>,
}
`;
